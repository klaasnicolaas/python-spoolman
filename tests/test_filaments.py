"""Test the Filament model for Spoolman."""

from __future__ import annotations

from typing import TYPE_CHECKING

from aresponses import ResponsesMockServer
from syrupy.assertion import SnapshotAssertion

from . import load_fixtures

if TYPE_CHECKING:
    from spoolman import Spoolman


async def test_get_filaments(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    spoolman_client: Spoolman,
) -> None:
    """Test getting all available filaments."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/filament",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("filaments.json"),
        ),
    )
    filaments = await spoolman_client.get_filaments()
    assert filaments == snapshot


async def test_get_filament(
    aresponses: ResponsesMockServer,
    snapshot: SnapshotAssertion,
    spoolman_client: Spoolman,
) -> None:
    """Test getting a specific filament."""
    aresponses.add(
        "192.168.x.x:7912",
        "/api/v1/filament/1",
        "GET",
        aresponses.Response(
            status=200,
            headers={"Content-Type": "application/json"},
            text=load_fixtures("single_filament.json"),
        ),
    )
    filament = await spoolman_client.get_filament(filament_id=1)
    assert filament == snapshot
