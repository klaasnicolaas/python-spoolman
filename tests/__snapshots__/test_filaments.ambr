# serializer version: 1
# name: test_get_filament
  Filament(id=1, name='Bambu Green', color='00AE42', vendor=Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc)), external_id='bambulab_pla_bambugreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=250.0, extruder_temp=220, bed_temp=60)
# ---
# name: test_get_filaments
  list([
    Filament(id=1, name='Bambu Green', color='00AE42', vendor=Vendor(id=1, name='Bambu Lab', external_id='Bambu Lab', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc)), external_id='bambulab_pla_bambugreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 3, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=250.0, extruder_temp=220, bed_temp=60),
    Filament(id=2, name='PolyTerraâ„¢ Forest Green', color='519f61', vendor=Vendor(id=2, name='Polymaker', external_id='Polymaker', registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc)), external_id='polymaker_pla_polyterraforestgreen_1000_175_n', registered=datetime.datetime(2024, 12, 28, 23, 32, 22, tzinfo=datetime.timezone.utc), material='PLA', density=1.24, diameter=1.75, weight=1000.0, spool_weight=140.0, extruder_temp=210, bed_temp=50),
  ])
# ---
